### DISCUSSION TITLE
How to scroll up after pagination?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#40613] [Help] [answered]
So, I have this piece of code here:

 ```
useMemo(() => {
    window.scrollTo({ top: 0 });
  }, [currentPage]);

```

^^  apparently this is not recognizing window...  and seems like its telling me I have unnecessary dependency??  but `currentPage` is a valid constant from which I want to run this Memo code... 

```
ReferenceError: window is not defined
  59 | 
  60 |   useMemo(() => {
> 61 |     window.scrollTo({ top: 0 });
     |    ^
  62 |   }, [currentPage]);

```

The full page code is this one:
https://github.com/Elindo586/nextprojectb/blob/main/pages/blog/castellano/index.js

**Comment by icyJoseph** [comment, marked as answer]
`useMemo` is used to store the result of a computation, generally one that's rather expensive to do, because of how Functional Components work, not gonna go into detail right now. 

> Think of it, for now, as a trade-off between time and memory. You pay with memory (storing the computation), in exchange for not spending time re-doing the computation, when there is a render pass, if the inputs are the same, that is.

What you want to do is `useEffect` to trigger a side-effect when `currentPage` changes, which is to scroll. We could also just do it on the handler that changes current page, just call the scrolling method immediately. 

However, in some cases perhaps we want to make sure the current page has indeed changed, and then do the scroll to top.

The reason why `window` throws is because when this runs on the server, the calculation is made (`useMemo`), but on the server there's no DOM, no window, so it crashes out.

```ts
useEffect(() => {
    window.scrollTo({ top: 0 });
  }, [currentPage]);
```

**Reply by Elindo586** [reply]
As always a wealth of knowledge and much help..... thanks!
